src/msalexample-ios-macos/Shared/MSALAuthentication.swift

```diff
@@ -0,0 +1,80 @@
+import MSAL
+
+class MSALAuthentication {
+    // 'Application (client) ID' of app registration in Azure portal - this value is a GUID
+    private static let kClientId = ""
+
+    // 'Tenant ID' of your Azure AD instance - this value is a GUID
+    private static let kTenantId = ""
+    
+    private static let kAuthority = try! MSALB2CAuthority(url: URL(string: "https://login.microsoftonline.com/\(kTenantId)")!)
+    private static let kConfig = MSALPublicClientApplicationConfig(clientId: kClientId, redirectUri: nil, authority: kAuthority)
+
+    // In order to take advantage of token caching, your MSAL client singleton must
+    // have a lifecycle that at least matches the lifecycle of the user's session in
+    // the app.
```

Comment:
```suggestion
    // To use token caching, your MSAL client singleton must have a lifecycle that
    // at least matches the lifecycle of the user's session in the app.
```

**Marsh's Writing Rule \#532**: Never, ever, ever use the words "in order" before "to." Just say "to" - the _in order_ superfluous. Just like "ever, ever" in this sentence. :wink: